baseplate/experiments/providers/simple_experiment.py

```diff
@@ -0,0 +1,208 @@
+        :param bucket_seed -- if provided, this provides the seed for determining
+            which bucket a variant request lands in. Providing a consistent
+            bucket_seed will ensure a user is bucketed consistently.
+        :param bool enabled -- whether or not this experiment is enabled.
+            disabling an experiment means all variant calls will return None
+        """
+
+        self.id = id
+        self.name = name
+        self.owner = owner
+        self.num_buckets = 1000
+
+        self.start_ts = start_ts
+        self.stop_ts = stop_ts
+        self.enabled = enabled
+
+        self.bucket_val = bucket_val
+        self.version = experiment_version
+        self.shuffle_version = shuffle_version
+        self.experiment_version = experiment_version
+
+        if not self.experiment_version:
+            raise ValueError('Experiment version must be provided.')
+
+        self.variants = variants
+
+        seed_data = {"id": id, "name": name, "shuffle_version": self.shuffle_version}
+        self._seed = bucket_seed or self.make_seed(seed_data)
+
+        self._validate_variants(self.variants)
+
+    @classmethod
+    def from_dict(cls, id, name, owner, start_ts, stop_ts, config,
+                  enabled=True, **kwargs):
+        bucket_val = config.get("bucket_val", "user_id")
+        version = config.get("experiment_version")
+        shuffle_version = config.get("shuffle_version")
+
+        variants = config.get("variants", [])
+
+        bucket_seed = config.get("bucket_seed")
+
+        return cls(
+            id=id,
+            name=name,
+            owner=owner,
+            start_ts=start_ts,
+            stop_ts=stop_ts,
+            enabled=enabled,
+            config=config,
+            experiment_version=version,
+            shuffle_version=shuffle_version,
+            variants=variants,
+            bucket_seed=bucket_seed,
+            bucket_val=bucket_val,
+        )
+
+    @property
+    def seed(self):
+        return self._seed
+
+    def get_unique_id(self, **kwargs):
+        if kwargs.get(self.bucket_val):
+            return ":".join(
+                [self.name, self.bucket_val, str(kwargs[self.bucket_val])]
+            )
+        else:
+            return None
+
+    def should_log_bucketing(self):
+        """ Default to true. Override if logging of eligibility events not required. """
+        return True
+
+    def variant(self, **kwargs):
+        lower_kwargs = {k.lower(): v for k, v in iteritems(kwargs)}
+
+        if not self._is_enabled():
+            return None
+
+        variant = self._check_overrides(**lower_kwargs)
+        if variant is not None and variant in self.variants:
+            return variant
+
+        if self.bucket_val not in lower_kwargs:
+            logger.info(
+                "Must specify %s in call to variant for experiment %s.",
+                self.bucket_val,
+                self.name,
+            )
+            return None
```

Comment:
nit: can you put a blank between these ifs please? when jammed together it reads from bird's eye view as an if/elif

Reply:
fixed